tests/lib/util/source-code.js

```diff
@@ -928,7 +961,110 @@ describe("SourceCode", () => {
+            const code = [
+                "var foo = {",
+                "    bar: 'bar'",
+                "    // comment",
+                "};",
+                "var baz;"
+            ].join("\n");
+
+            eslint.reset();
+            eslint.on("Program", assertCommentCount(0, 0));
+            eslint.on("VariableDeclaration", assertCommentCount(0, 0));
+            eslint.on("VariableDeclerator", assertCommentCount(0, 0));
+            eslint.on("Identifier", assertCommentCount(0, 0));
+            eslint.on("ObjectExpression", assertCommentCount(0, 0));
+            eslint.on("Property", assertCommentCount(0, 1));
+            eslint.on("Literal", assertCommentCount(0, 0));
+
+            eslint.verify(code, config, "", true);
+        });
+
+        it("should return comments for an export default anonymous class", () => {
+            const code = [
+                "/**",
+                " * this is anonymous class.",
+                " */",
+                "export default class {",
+                "    /**",
+                "     * this is method1.",
+                "     */",
+                "    method1(){",
+                "    }",
+                "}"
+            ].join("\n");
+
+            eslint.reset();
+            eslint.on("Program", assertCommentCount(0, 0));
+            eslint.on("ExportDefaultDeclaration", assertCommentCount(1, 0));
+            eslint.on("ClassDeclaration", assertCommentCount(0, 0));
+            eslint.on("ClassBody", assertCommentCount(0, 0));
+            eslint.on("ClassBody", assertCommentCount(0, 0));
+            eslint.on("MethodDefinition", assertCommentCount(1, 0));
+            eslint.on("Identifier", assertCommentCount(0, 0));
+            eslint.on("FunctionExpression", assertCommentCount(0, 0));
+            eslint.on("BlockStatement", assertCommentCount(0, 0));
+
+            eslint.verify(code, config, "", true);
+        });
+
+        it("should return leading comments", () => {
+            let varDeclCount = 0;
+            const code = [
+                "//#!/usr/bin/env node",
+                "var a;",
+                "// foo",
+                "var b;",
+            ].join("\n");
+
+            eslint.reset();
+            eslint.on("Program", assertCommentCount(0, 0));
+            eslint.on("VariableDeclaration", node => {
+                if (varDeclCount === 0) {
+                    assertCommentCount(1, 1)(node);
+                } else {
+                    assertCommentCount(1, 0)(node);
+                }
+                varDeclCount++;
+            });
+            eslint.on("VariableDeclarator", assertCommentCount(0, 0));
+            eslint.on("Identifier", assertCommentCount(0, 0));
+            eslint.on("VariableDeclarator", assertCommentCount(0, 0));
+            eslint.on("Identifier", assertCommentCount(0, 0));
+
+            eslint.verify(code, config, "", true);
+        });
```

Comment:
Could you please add a test for a multiple-declarator VariableDeclaration, so we understand how the comment attachment is supposed to work there?

Example test case:

```js
// Leading comment for VariableDeclaration?
var a,  // Trailing comment for VariableDeclarator? And/or leading for the next?
    b,  // Trailing comment for second VariableDeclarator?
    c;  // Trailing comment for VariableDeclaration?
// Trailing comment for VariableDeclaration?
```